src/agents/atomdb_broker/AtomDBProxy.cc

```diff
@@ -118,6 +119,24 @@ vector<string> AtomDBProxy::add_atoms(const vector<Atom*>& atoms, bool use_strea
     return handles;
 }
 
+uint AtomDBProxy::delete_atoms(const vector<string>& handles, bool delete_link_targets) {
+    vector<string> args;
+    // split handles into chunks to avoid exceeding command size limits
+    size_t chunk_size = AtomDBProxy::COMMAND_SIZE_LIMIT;
+    uint total_deleted = 0;
+    for (size_t i = 0; i < handles.size(); i += chunk_size) {
+        args.clear();
+        size_t end = std::min(i + chunk_size, handles.size());
+        args.insert(args.end(), handles.begin() + i, handles.begin() + end);
+        args.push_back(delete_link_targets ? "1" : "0");
+        Utils::retry_function([&]() { this->to_remote_peer(AtomDBProxy::DELETE_ATOMS, args); },
```

Comment:
This should be moved up to the proxy superclass so all proxies could share the same behavior.